FEATURE:- Milestone6(Friendships V2)

vanheaven-ui opened the pull request:
Hi, 
In this PR:
- [x] Additional User/Friendship `has many, through` associations were defined in the User model.
- [x] Methods in the User model were refactored based on the newly defined associations.
- [x] View logic was moved to the friendship helper module.
- [x] Rspec unit tests were written for the models.

The app features in this PR include; **Sending friend requests**, **accepting friend requests**, **rejecting friend requests**, and **canceling friendships**. 

**Note:** Running rspec tests might require running `bundle exec rspec` command in the terminal if running `rspec` fails

vanheaven-ui committed:
Modify Friendship associations

Add through user/friendship  associations

Refactor user model methods

vanheaven-ui committed:
Move view logic to friendship helper module

Resolve rubocop offenses

vanheaven-ui committed:
Add destroy and decline actions

vanheaven-ui committed:
Add find_either_friendship method

vanheaven-ui committed:
Add decline_request, cancel_friend routes

vanheaven-ui committed:
Add signed in current user for identification

vanheaven-ui committed:
Add helper function for view logic

vanheaven-ui committed:
Modify user lists

vanheaven-ui committed:
Amend menu class width attribute to 50%

vanheaven-ui committed:
Resolve rubocop offenses

vanheaven-ui committed:
Add unit tests

Resolve rubocop offenses

christianotieno reviewed (changes requested):
# Changes Required ♻️ 

Hello team 👋 
Good job with everything. Things look great. Keep it up. 👍 
- [ ] There are some missing dependencies that are affecting the running of the RSpec tests. Kindly before we proceed, add them to the general gem files to make sure our tests can run anywhere locally, thanks. 

### Happy coding 💻 

#### Christian Otieno (TSE)

vanheaven-ui committed:
Modify Gemfile Rspec dependencies

addod19 reviewed (approved):
## Status: Approved :trophy: :balloon: 

Hi @vanheaven-ui, congratulations on a great done:1st_place_medal: Having fixed the comment outlined by the previous TSE :+1: I hereby approve of the project :smile: 
Go ahead and merge into development :100: 

See you at the top :airplane: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

vanheaven-ui committed:
Add timeline count of likes and comment

Add margin between accept and decline links

Add color scheme to accept and reject links

vanheaven-ui committed:
Add screenshot and reslove stylelint errors

vanheaven-ui committed:
Resolve invalid value for color:
